feat(array): BTreeMap conversion

kakserpom opened the pull request:
## Description

I've added the same `FromZval`/`IntoZval` conversions for `BTreeMap` that exist for `HashMap`.

I know a doc update wouldn't hurt, but I am too preoccupied with development atm.

Xenira commented:
Also could you elaborate on this? https://github.com/davidcole1340/ext-php-rs/pull/520#discussion_r2217875652

kakserpom commented:
@Xenira 

> Also could you elaborate on this? [#520 (comment)](https://github.com/davidcole1340/ext-php-rs/pull/520#discussion_r2217875652)

I've implemented your suggestion. I've also fixed the cause of `clippy::implicit_hasher` warning in `hash_map` by generalizing the hasher.

kakserpom commented on src/types/array/conversions/mod.rs:
```diff
@@ -9,5 +9,6 @@
 //! - `HashMap<K, V>` ↔ `ZendHashTable` (via `hash_map` module)
 //! - `Vec<T>` and `Vec<(K, V)>` ↔ `ZendHashTable` (via `vec` module)
```
Done.

kakserpom commented:
> LGTM but would it be possible to have some tests?

What's there to test? Rustc does the job pretty well :)

ptondereau commented:
> > LGTM but would it be possible to have some tests?
> 
> What's there to test? Rustc does the job pretty well :)

Just ensuring the conversion if it makes sense

kakserpom commented:
And also I've moved the code that treats numeric string keys as` long`, now it happens in ArrayKey conversion impls. So no more boilerplate in `impl ZendHashTable`.
